src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -758,10 +759,41 @@ private static bool IsAllPunctuation(string filterText)
         /// </summary>
         private static bool IsPotentialFilterCharacter(char c)
         {
-            // TODO(cyrusn): Actually use the right unicode categories here.
+            // TODO(cyrusn): Actually use the right Unicode categories here.
             return char.IsLetter(c)
                 || char.IsNumber(c)
                 || c == '_';
         }
+
+        /// <summary>
+        /// This comparer can handle expanded items.
+        /// </summary>
+        private class CompletionItemSortingComparer : IComparer<VSCompletionItem>
+        {
+            public static CompletionItemSortingComparer Instance { get; } = new CompletionItemSortingComparer();
+
+            public int Compare(VSCompletionItem x, VSCompletionItem y)
+            {
```

Comment:
why can't this just check `CompletionItem.Flag`?

Reply:
Because the flag is on Roslyn CompletionItem, to get that we will need to get it from the VS item property bag first. We can put the flag in VS item's property bag directly, but I'm not sure whether it will be a perf hit with 10s of thousands items. Need to do some profiling

Reply:
Doesn't the VSCompletionItem already know this info?  After all, doesn't it have this in its filters so it knows what is an 'expanded item' and what isn't?

Reply:
It does, but the info is in an 'ImmutableArray<CompletionFilter>'. We will need to search though the list every time a comparison is made. As I said about retrieving Roslyn item from the property bag, it might or might not be a perf issue, just need to check.

Reply:
The profiling result doesn't show much difference if we get the Roslyn item from the property bag and just compare that, so I made the change as you suggested above.